### PR TITLE
Add a note to the Baggage API to clarify that languages are free to use language-friendly idiom to implement the API.

### LINKED ISSUE #1051: The Baggage API is not consistent with other APIs
**What are you trying to achieve?**
Implement the Baggage API in such a way that it's consistent with the other OTel APIs

What did you expect to see?
I expected the Baggage to be a thing that could be extracted, wholesale out of the Context, manipulated via an extra-Context API, then put back into the Context (if needed). Instead, the API requires all key/value interactions with the Baggage to be via the Context. So, to read and add some keys to the baggage, you have to do something like:

```java
Entry entry = BaggageContextUtils.getValue(context, "foo");
Context clearedContext = BaggageContextUtils.clearBaggage(context);
Context newContext = BaggageContextUtils.addValue(clearedContext, "foo", new Entry("stuff"));
Context yetAnotherNewContext = BaggageContextUtils.addValue(newContext, "boo", new Entry("other stuff"));
```

I was expecting to provide something like this, in Java:
```java
Baggage baggage = BaggageContextUtils.getBaggage(context);
Entry entry = baggage.getEntry("foo");
Baggage newBaggage = new Baggage();
newBaggage.setEntry("baz", new Entry("stuff"));
newBaggage.setEntry("boo", new Entry("other stuff"));
Context newContext = BaggageContextUtils.setBaggage(context, baggage);
```

These APIs are functionally equivalent, but it feels quite awkward to manipulate individual keys coming off the wire via the Context, rather than deal with Baggage as a unified construct.

**Additional context.** 

Current Baggage API spec is here:  https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/baggage/api.md

### PR DESCRIPTION
**Pull request by jkwatson**
Fixes #1051

## Changes

Updates the Baggage API to clarify that the Baggage may be implemented with an immutable struct/object.

### CODE REVIEW

File: specification/baggage/api.md [hunk 1]
```diff
@@ -24,6 +24,15 @@ The Baggage API consists of:
 - the `Baggage`
 - functions to interact with the `Baggage` in a `Context`
 
+Please note: the functions described here are one way to approach interacting with the Baggage
```
Comment (arminru) [inline specification/baggage/api.md:27]:
```suggestion
The functions described here are one way to approach interacting with the Baggage
```

If it contains normative statements (like the MUST below), it should not be stated as a "note".

File: specification/baggage/api.md [hunk 2]
```diff
@@ -24,6 +24,15 @@ The Baggage API consists of:
...
+purely via the Context. Depending on language idioms, a language API may implement these functions
```
Comment (arminru) [inline specification/baggage/api.md:28]:
```suggestion
purely via the Context. Depending on language idioms, a language API MAY implement these functions
```

Comment (jkwatson) [reply specification/baggage/api.md:28]: I didn't really intend this one to be a normative MAY, but I'm ok with that if you prefer it.

Comment (arminru) [reply specification/baggage/api.md:28]: Well it explicitly allows SIGs to diverge from the API specified below, so I think it should be marked as such. As far as I know, we don't have any general statement permitting to diverge from the spec in order to follow language-specifics but usually consider and mention that on a per-case basis.

**Review by anuraaga — APPROVED** [review]

**Comment by carlosalberto** [comment]
@bogdandrutu Would you mind give this your blessing? ;)

**Review by bogdandrutu — APPROVED** [review]
I still believe that the spec should be written with the immutable Baggage object, to be consistent with the Span, but I am fine for the moment.

**Comment by carlosalberto** [comment]
@bogdandrutu Created #1094 to track this one ;)

**Pull request merged**